compiler/modules/front.syntax/ast.bal

```diff
@@ -472,7 +472,17 @@ public type FieldDesc record {|
     TypeDesc typeDesc;
 |};
 
-public type MappingTypeDesc record {|
+public type MappingTypeDesc MapTypeDesc|RecordTypeDesc;
+
+public type MapTypeDesc record {|
+    *PositionFields;
+    TypeDesc typeParam;
+    t:MappingDefinition? defn = ();
+|};
+
+public type RecordTypeDesc ExclusiveRecordTypeDesc|InclusiveRecordTypeDesc;
```

Comment:
I think it's simpler to handle this by changing `rest` have type TypeDesc|boolean, where true is used for an inclusive descriptor.

Reply:
fixed